### PR TITLE
Add tests for PaymentRequest .abort() method

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@rsolomakhin, these are incoming... All green in Chrome 🤘

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/236254025)

# Firefox (nightly) #
Testing web-platform-tests at revision 082edec7bb5c1d756253e13400f7973de35f4402
Using browser at version BuildID 20170524100215; SourceStamp 291a11111bdd05c5cd55dd552da4b1285ceba9b2
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/6061/payment-request/payment-request-abort-method.https.html">/payment-request/payment-request-abort-method.https.html</a></summary>


|                                  Subtest                                  | Results |                                                Messages                                                |
|---------------------------------------------------------------------------|---------|--------------------------------------------------------------------------------------------------------|
|                                                                           | OK      |                                                                                                        |
| `Throws if the promise [[state]] is not "interactive"`                    | FAIL    | `promise_test: Unhandled rejection with value: object "ReferenceError: PaymentRequest is not defined"` |
| `Calling abort must not change the [[state]] until after interactive`     | FAIL    | `promise_test: Unhandled rejection with value: object "ReferenceError: PaymentRequest is not defined"` |
| `calling .abort() causes acceptPromise to reject and closes the request.` | FAIL    | `promise_test: Unhandled rejection with value: object "ReferenceError: PaymentRequest is not defined"` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/236254026)

# Sauce (safari) #
Testing web-platform-tests at revision 082edec7bb5c1d756253e13400f7973de35f4402
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="https://w3c-test.org/submissions/6061/payment-request/payment-request-abort-method.https.html">/payment-request/payment-request-abort-method.https.html</a></summary>


|                 Subtest                  | Results |                                     Messages                                    |
|------------------------------------------|---------|---------------------------------------------------------------------------------|
|                                          | OK      |                                                                                 |
| `Test for PaymentRequest.abort() method` | FAIL    | `SyntaxError: Unexpected identifier 't'. Expected ')' to end an argument list.` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/236254032)

# Chrome (unstable) #
Testing web-platform-tests at revision d3aa5f9775b6e0cd1b7703542a61d1fa9f7d7dae
Using browser at version 60.0.3107.4 dev
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/236254033)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision d3aa5f9775b6e0cd1b7703542a61d1fa9f7d7dae

**Comment by marcoscaceres** [comment]
@rsolomakhin you might like these also. Can you review them? 
https://w3c-test.org/submissions/6061/payment-request/payment-request-abort-method.https.html

File: payment-request/payment-request-abort-method.https.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+<!-- Copyright © 2017 Chromium authors and World Wide Web Consortium, (Massachusetts Institute of Technology, ERCIM, Keio University, Beihang). -->
+<meta charset="utf-8">
+<title>Test for PaymentRequest.abort() method</title>
+<link rel="help" href="https://w3c.github.io/browser-payment-api/#abort-method">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+"use strict";
+setup(() => {}, {
+  // Ignore unhanded rejections resulting from .show()'s acceptPromise
```
Comment (sideshowbarker) [inline payment-request/payment-request-abort-method.https.html]: >     +  // Ignore unhanded rejections resulting from .show()'s acceptPromise

Nit: unhanded ➡️ unhandled

File: payment-request/payment-request-abort-method.https.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  request.show();
+  // Let's set request the state to "closed" by calling .abort()
+  try {
+    await request.abort();
+  } catch (err) {
+    assert_true(false, "Unexpected promise rejection: " + err.message);
+  }
+  // The request is now "closed", so...
+  await promise_rejects(t, "InvalidStateError", request.abort());
+}, "Calling abort must not change the [[state]] until after interactive");
```
Comment (sideshowbarker) [inline payment-request/payment-request-abort-method.https.html]: >      +}, "Calling abort must not change the [[state]] until after interactive");

Minor-minor nit: This test description might be more clear with *interactive* in quotes: `until after 'interactive'`

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**